get_github_coderabbit_review/coderabbit_fetcher/cache/redis_cache.py

```
@@ -0,0 +1,401 @@
+            else:
+                return json.dumps(data, default=str)
+
+        except Exception as e:
+            logger.error(f"Error serializing cache entry: {e}")
+            raise CacheError(f"Serialization failed: {e}")
+
+    def _deserialize_entry(self, data: Union[str, bytes]) -> CacheEntry:
+        """Deserialize cache entry from storage."""
+        try:
+            if self.config.serialization == "pickle":
+                entry_data = pickle.loads(data)
```

Ruff (0.13.1)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)